packages/ppx/src/declarations_to_string.re

```diff
@@ -0,0 +1,425 @@
+open Ppxlib;
+open Reason_css_parser;
+
+module Helper = Ast_helper;
+module Builder = Ppxlib.Ast_builder.Default;
+
+/* helpers */
+let loc = Location.none;
+let txt = txt => {Location.loc: Location.none, txt};
+
+let (let.ok) = Result.bind;
+
+exception Unsupported_feature;
+
+let id = Fun.id;
+
+let render_string = string =>
+  Helper.Const.string(string) |> Helper.Exp.constant;
+
+let render_integer = integer =>
+  Helper.Const.int(integer) |> Helper.Exp.constant;
+
+type transform('ast, 'value) = {
+  ast_of_string: string => result('ast, string),
+  value_of_ast: 'ast => 'value,
+  value_to_expr: 'value => list(Parsetree.expression),
+  ast_to_expr: 'ast => list(Parsetree.expression),
+  string_to_expr: string => result(list(Parsetree.expression), string),
+};
+
+let emit = (property, value_of_ast, value_to_expr) => {
+  let ast_of_string = Parser.parse(property);
+  let ast_to_expr = ast => value_of_ast(ast) |> value_to_expr;
+  let string_to_expr = string =>
+    ast_of_string(string) |> Result.map(ast_to_expr);
+
+  {ast_of_string, value_of_ast, value_to_expr, ast_to_expr, string_to_expr};
+};
+
+let variants_to_string =
+  fun
+  | `Row => id([%expr "row"])
```

Comment:
A little random idea: What about generating this render methods from the Parser.re ppx? I think there we have all the information needed to turn from variants/combinations of values into strings already. Nothing that we want to do now, but maybe worth considering? WDYT?

Reply:
I think that it makes a lot of sense, it would be easier to just add more properties and it would remove the necessity to write a lot of  similar functions to `variant_to_string`